### PR TITLE
Factor gzip/deflate code

### PR DESCRIPTION
**Pull request by jankeromnes**
@espadrine please have a look. You pretty much already reviewed this :)

### CODE REVIEW

File: lib/camp.js [hunk 1]
```diff
@@ -156,6 +156,22 @@ function getQueries(ask, end) {
...
+function getCompressedStream(ask) {
+  var encoding = ask.req.headers['accept-encoding'] || '';
+  var res = ask.res;
+  if (encoding.match(/\bgzip\b/)) {
+    ask.res.setHeader('content-encoding', 'gzip');
+    res = zlib.createGzip();
+    res.pipe(ask.res);
+  } else if (encoding.match(/\bdeflate\b/)) {
+    ask.res.setHeader('content-encoding', 'deflate');
+    res = zlib.createGzip();
```
Comment (espadrine) [inline lib/camp.js]: Did you mean `zlib.createDeflate()`?

:clippy:

Comment (jankeromnes) [inline lib/camp.js]: omg how did that happen?

**Comment by espadrine** [comment]
Thanks!

**Comment by jankeromnes** [comment]
:+1:

**Pull request merged**